kvstore/kvstore_cm.go

```diff
@@ -0,0 +1,123 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Simple abstraction for storing state on a k8s ConfigMap. Very very simple
+// and uses a single entry in the ConfigMap.data for storing serialized
+// JSON of the generic data that Load/Save uses. Handy for things like sources
+// that need to persist some state (checkpointing for example).
+package kvstore
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"knative.dev/pkg/logging"
+)
+
+type ConfigMapKVStore struct {
+	cmClient  v1.ConfigMapInterface
+	name      string
+	namespace string
+	data      map[string]string
+}
+
+var (
+	_ KVStore = (*ConfigMapKVStore)(nil)
+)
+
+func NewConfigMapKVStore(name string, namespace string, clientset v1.CoreV1Interface) *ConfigMapKVStore {
+
+	return &ConfigMapKVStore{name: name, namespace: namespace, cmClient: clientset.ConfigMaps(namespace)}
+}
+
+// Init initializes ConfigMapKVStore either by loading or creating an empty one.
+func (cs *ConfigMapKVStore) Init(ctx context.Context) error {
```

Comment:
It would be nice to a) have Init in another interface next to `KVStore` or b) add Init to `KVStore`